File: src/main/java/dev/cernavskis/moose/util/DebugInfo.java
package dev.cernavskis.moose.util;

import dev.cernavskis.moose.lexer.Token;

public record DebugInfo(int line, int column, String file, Token token) {
  @Override
  public String toString() {
    return "DebugInfo{" +
      "line=" + line +
      ", column=" + column +
      ", file='" + file + '\'' +
      ", token=" + token +
      '}';
  }
}

File: src/util/debug_info.rs
use crate::lexer::Token;

#[derive(Debug, Clone)]
pub struct DebugInfo {
    pub line: usize,
    pub column: usize,
    pub file: String,
    pub token: Option<Token>,
}

impl DebugInfo {
    pub fn new(line: usize, column: usize, file: String, token: Option<Token>) -> Self {
        DebugInfo { line, column, file, token }
    }
}
